includes/Services/SiteGenService.php

```diff
@@ -969,6 +971,29 @@ public static function publish_sitemap_pages( $site_description, $content_style,
 		return true;
 	}
 
+	/**
+	 * Checks if it is a products or shop slug
+	 *
+	 * @param string $slug Slug of the page being generated.
+	 * @return boolean
+	 */
+	public static function is_products_or_shop_page( $slug ) {
+		return ( 'products' === $slug || 'shop' === $slug );
+	}
+
+	/**
+	 * Sets the products page generated by AI to woocommerce shop page
+	 *
+	 * @param integer $page_id Id of the page
+	 */
+	public static function set_woo_shop_page( $page_id ) {
+		$prev_page_id = Options::get_option_name( 'woocommerce_shop_page_id', false );
```

Comment:
Please check this line of code. I think that you are expecting the page_id but are retrieving the option_name, so `if` will always be true.

In what case will we get the same page_id again? Do we need this check?